### PR TITLE
Add `empty_result_state` attribute to the `databricks_sql_alert` resource

### LINKED ISSUE #2662: [FEATURE] Support default DBSQL alert state when query returns empty results
### Use-cases
Configure default DBSQL alert state when query returns empty rows.

### Proposal
For `databricks_sql_alert` resource, support `empty_result_state` under `option` for default DBSQL alert state when query returns empty rows. This is already supported in Databricks API: https://github.com/databricks/universe/pull/380668.

### PR DESCRIPTION
**Pull request by alexott**
## Changes

This fixes #2662

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] tested manually
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: sql/resource_sql_alerts.go [hunk 1]
```diff
@@ -46,14 +47,18 @@ func (a *AlertEntity) toCreateAlertApiObject(s map[string]*schema.Schema, data *
 		Op:            a.Options.Op,
 		Value:         a.Options.Value,
 	}
-
-	return ca, nil
+	// This is a workaround for Go SDK problem, will be fixed there.
+	var err error
+	if a.Options.EmptyResultState != "" {
+		err = ca.Options.EmptyResultState.Set(a.Options.EmptyResultState)
```
Comment (tanmay-db) [inline sql/resource_sql_alerts.go:53]: We should throw an error if the value is not supported

> Currently supported values are `unknown`, `triggered`, `ok`

File: sql/resource_sql_alerts.go [hunk 2]
```diff
@@ -46,14 +47,18 @@ func (a *AlertEntity) toCreateAlertApiObject(s map[string]*schema.Schema, data *
 		Op:            a.Options.Op,
 		Value:         a.Options.Value,
 	}
-
-	return ca, nil
+	// This is a workaround for Go SDK problem, will be fixed there.
```
Comment (tanmay-db) [inline sql/resource_sql_alerts.go:50]: Which problem is this?

**Review by tanmay-db — APPROVED** [review]

File: sql/resource_sql_alerts.go (hunk 2 above)
Comment (alexott) [reply sql/resource_sql_alerts.go:50]: The problem with SDK that it doesn't accept empty string for optional enums

File: sql/resource_sql_alerts.go (hunk 1 above)
Comment (alexott) [reply sql/resource_sql_alerts.go:53]: The check should be done on the backend. The main problem right now is that I can't simply do `ca.Options.EmptyResultState.Set(a.Options.EmptyResultState)` if the value is omitted - SDK returns an error.

File: sql/resource_sql_alerts.go (hunk 2 above)
Comment (alexott) [reply sql/resource_sql_alerts.go:51]: Tracked here: https://github.com/databricks/databricks-sdk-go/issues/630

**Comment by junwei-db** [comment]
@alexott I noticed the documentation for this change is incorrect -- the new `empty_result_state` attribute should be specified under `databricks_sql_alert.options`, not directly under `databricks_sql_alert`. Could you help fix this? Thanks

**Comment by alexott** [comment]
@junwei-db it's just a doc problem, implementation is in the correct place

**Comment by junwei-db** [comment]
Right, that's what I was referring to. Could you help fix the doc?

**Pull request merged**